torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -291,6 +292,9 @@ static inline bool treatSequenceAsTuple(PyObject* index) {
   if (n >= 32) {
     return false;
   }
+  if (PyArray_CheckExact(index)) {
+    return false;
+  }
```

Comment:
You cannot do this unconditionally, as NumPy is not a runtime dependency of PyTorch. You need to do this just if `USE_NUMPY` is true. You also need to initialize NumPy, by checking that
```python
  TORCH_CHECK(is_numpy_available(), "Numpy is not available");
```
as it may not be initialised, in which case `CheckExact` may break.

Also, could you move this up, next to the tensor check and update the  comment to note that we also consider NumPy arrays?